src/main/webapp/i18n/de/course.json

```diff
@@ -90,6 +90,27 @@
             "upload": "Hochladen",
             "uploading": "Lade hoch...",
             "fileUploadExercises": "Dateiupload-Aufgabe",
+            "search": {
+                "searchButton": "Suchen",
+                "toggleButton": {
+                    "true": "Zeige erweiterte Suche",
+                    "false": "Verstecke erweiterte Suche"
+                },
+                "cardTitle": "Aufgaben durchsuchen",
+                "fieldExerciseName": "Aufgaben nach Namen durchsuchen",
+                "labelExerciseCategory": "Kategorie",
+                "fieldExerciseCategory": "Kategorie",
+                "labelExerciseType": "Aufgabentyp",
+                "noResults": "Keine Aufgaben wurden für die Suchparameter gefunden",
+                "type": {
+                    "all": "Alle Aufgaben",
+                    "programming": "Programmieraufgaben",
+                    "quiz": "Quizaufgaben",
+                    "modeling": "Modellierungsaufgaben",
+                    "text": "Textaufgaben",
+                    "file-upload": "Dateiupload-Aufgaben"
```

Comment:
```suggestion
                    "file-upload": "Dateiuploadaufgaben"
```

Reply:
While I must agree with you that this would make them consistent, this is not how it is done in the rest of Artemis where they are also called "Dateiupload-Aufgabe" (Just take a look above the "search" block I added in course.json.
So I'll leave it as it is to keep it in line with the other translations.